velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -220,7 +224,13 @@ void ReaderBase::initializeSchema() {
   // is the root itself. This is ok because it's never required to check the
   // parent of the root in getParquetColumnInfo().
   schemaWithId_ = getParquetColumnInfo(
-      maxSchemaElementIdx, maxRepeat, maxDefine, 0, schemaIdx, columnIdx);
+      maxSchemaElementIdx,
+      maxRepeat,
+      maxDefine,
+      0,
+      schemaIdx,
+      columnIdx,
+      options_.fileSchema());
```

Comment:
Let's use `fileType` if table schema (`Options::fileSchema`) is missing

Reply:
With this you should be able to remove getRequestedType. You can just add a `VELOX_CHECK` inside `getParquetColumnInfo`

Reply:
@kevincmchen Is this comment addressed?